src/main/java/io/numaproj/numaflow/info/ContainerType.java

```diff
@@ -0,0 +1,40 @@
+package io.numaproj.numaflow.info;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * the string content matches the corresponding server info file name.
+ * DO NOT change it unless the server info file name is changed.
+ */
+public enum ContainerType {
+    SOURCER("sourcer"),
+    SOURCE_TRANSFORMER("sourcetransformer"),
+    SINKER("sinker"),
+    MAPPER("mapper"),
+    REDUCER("reducer"),
+    REDUCE_STREAMER("reducestreamer"),
+    SESSION_REDUCER("sessionreducer"),
+    SIDEINPUT("sideinput"),
+    FBSINKER("fb-sinker"),
+    UNKNOWN("unknown");
+
+    private final String name;
+
```

Comment:
Do we need to support `UNKNOWN`?

Reply:
No, I put it mainly for error scenarios.